bin/reth/src/commands/node/mod.rs

```diff
@@ -128,6 +132,77 @@ pub struct NodeCommand<Ext: RethCliExt = ()> {
 }
 
 impl<Ext: RethCliExt> NodeCommand<Ext> {
+    /// Launches the node
+    ///
+    /// This transforms the node command into a node config and launches the node using the given
+    /// closure.
+    pub async fn execute2<L, Fut>(self, ctx: CliContext, launcher: L) -> eyre::Result<()>
+    where
+        L: FnOnce(WithLaunchContext<Arc<DatabaseEnv>, InitState>, Ext::Node) -> Fut,
+        Fut: Future<Output = eyre::Result<()>>,
+    {
+        let Self {
+            datadir,
+            config,
+            chain,
+            metrics,
+            trusted_setup_file,
+            instance,
+            with_unused_ports,
+            network,
+            rpc,
+            txpool,
+            builder,
+            debug,
+            db,
+            dev,
+            pruning,
+            #[cfg(feature = "optimism")]
+            rollup,
+            ext,
+        } = self;
+
+        let data_dir = datadir.unwrap_or_chain_default(chain.chain);
+        let db_path = data_dir.db_path();
+
+        tracing::info!(target: "reth::cli", path = ?db_path, "Opening database");
+        let database = Arc::new(
+            init_db(db_path.clone(), DatabaseArguments::default().log_level(db.log_level.clone()))?
+                .with_metrics(),
+        );
+
+        // set up node config
+        let mut node_config = NodeConfig {
+            // TODO
+            database: DatabaseBuilder::Test,
```

Comment:
guessing the TODO is for setting this to `DatabaseBuilder::Real(db)`

Reply:
no removing it